doc/user_guide/sparse_backend.rst

```diff
@@ -0,0 +1,149 @@
+memory than the dense version (``tensorly.decomposition.parafac``) for a
+sparse input tensor.
+
+Missing Values
+~~~~~~~~~~~~~~
+
+The ``parafac`` function supports masks for missing values. A mask should be a
+boolean array of the same shape as the original tensor that is ``False``/``0``
+where the value is missing and ``True``/``1`` where it is not. It is passed to
+the ``parafac()`` function via the ``mask`` parameter.
+
+When using masks with sparse tensors, there are two important caveats:
+
+- The mask itself should be sparse. In the algorithm, the memory used will be
+  proportional to the number of missing values. Values of the original sparse
+  tensor that are ``0`` should be ``True``/``1`` (i.e., non-missing) in the
+  mask.
+
+- Sparse mask support is currently only supported with the sparse backend
+  ``parafac`` (``tensor.contrib.sparse.decomposition.parafac``). The dense
+  backend ``parafac`` will densify the array.
+
+Example
+=======
+
+In the following example, we construct a random sparse tensor that has a known
+rank (by construction), and decompose it with parafac.
+
+.. code:: python
+
+   >>> shape = (1000, 1001, 1002)
+   >>> rank = 5
+   >>> starting_factors = [sparse.random((i, rank)) for i in shape]
+   >>> starting_factors
+   [<COO: shape=(1000, 5), dtype=float64, nnz=50, fill_value=0.0>, <COO: shape=(1001, 5), dtype=float64, nnz=50, fill_value=0.0>, <COO: shape=(1002, 5), dtype=float64, nnz=50, fill_value=0.0>]
+
+Here we construct a tensor from the random factors. Note that in general,
+a recomposed tensor will be dense, but for our constructed example it is
+sparse, so we can use ``kruskal_to_tensor`` without worrying about using too
+much memory.
+
+   >>> from tensorly.contrib.sparse.kruskal_tensor import kruskal_to_tensor
+   >>> tensor = kruskal_to_tensor(starting_factors)
+   >>> tensor
+   <COO: shape=(1000, 1001, 1002), dtype=float64, nnz=5044, fill_value=0.0>
+
+This is how much memory the sparse array takes up, vs. how much it would take
+up if it were represented densly.
+
+   >>> tensor.nbytes / 1e9 # Actual memory usage in GB
+   0.000161408
+   >>> import numpy as np
+   >>> np.prod(tensor.shape) * 8 / 1e9 # Memory usage if array was dense, in GB
+   8.024016
+
+Now to decompose the tensor.
+
+   >>> from tensorly.decomposition import parafac # The dense version
+   >>> import time
+   >>> t = time.time(); factors = parafac(tensor, 5, init='random'); print(time.time() - t)
+   1.3858051300048828
+
+Note that the decomposition takes much longer when using the sparse variant.
+
+   >>> from tensorly.contrib.sparse.decomposition import parafac as sparse_parafac # The sparse version
+   >>> t = time.time(); sparse_factors = sparse_parafac(tensor, 5, init='random'); print(time.time() - t)
+   14.053689002990723
+
+However, there can be advantages to using the sparse variant. It is currently
+required when using ``init='svd'`` to make TensorLy use the sparse SVD
```

Comment:
I had thought it was also required to use a mask, but it seems the code has changed https://github.com/tensorly/tensorly/pull/105/files#r274059484. Was the sparse `kr` with a mask not implemented correctly?

Reply:
I'm not sure I get this -- are you saying we should specify the mask only works for sparse backend?

Reply:
The version I originally wrote only worked for the sparse wrapped `parafac`. I don't know if that is still the case after the changes that were made to `kr` and friends.